test/src/edu/stanford/nlp/ling/WordLemmaTagTest.java

```diff
@@ -0,0 +1,69 @@
+package edu.stanford.nlp.ling;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class WordLemmaTagTest {
+
+  private void checkWordLemmaTag(WordLemmaTag wLT, String word, String lemma, String tag) {
+    assertEquals("Incorrect word", word, wLT.word());
+    assertEquals("Incorrect lemma", lemma, wLT.lemma());
+    assertEquals("Incorrect tag", tag, wLT.tag());
+  }
+
+  @Test
+  public void testWordLemmaConstructorLabelWord() {
+    WordLemmaTagFactory wLtF = new WordLemmaTagFactory();
+    Label label = wLtF.newLabel("running");
+
+    WordLemmaTag wLt = new WordLemmaTag(label);
+
+    checkWordLemmaTag(wLt, "running", null, null);
+  }
+
+  @Test
+  public void testWordLemmaConstructorLabelWordLabelTag() {
+    WordLemmaTagFactory wLtF = new WordLemmaTagFactory();
+    Label labelWord = wLtF.newLabel("running");
+    Label labelTag = wLtF.newLabel("r");
+
+    WordLemmaTag wLt = new WordLemmaTag(labelWord, labelTag);
+
+    checkWordLemmaTag(wLt, "running", "running", "r");
+  }
+
+  @Test
+  public void testWordLemmaConstructorWordTag() {
+    String word = "run";
+    String tag = "r";
+    WordLemmaTag wLt = new WordLemmaTag(word, tag);
+
+    checkWordLemmaTag(wLt, "run", "run", "r");
+  }
+
+  @Test
+  public void testSetFromStringWordAndLemma() {
+    WordLemmaTag wLT = new WordLemmaTag();
+    wLT.setFromString("running/r");
+
+    checkWordLemmaTag(wLT, "running", "running", "r");
+  }
+
+  @Test
+  public void testSetFromStringWordAndLemmaAndTag() {
+    WordLemmaTag wLT = new WordLemmaTag();
+    wLT.setFromString("studying/study/s");
+
+    checkWordLemmaTag(wLT, "studying", "study", "s");
+  }
+
+///// Failing TEST
```

Comment:
Failing, in the sense that you expect it to have null values when set this way?

Reply:
The constructor of class WordLemaTag supports setting only the word value`  public WordLemmaTag(Label word) {
    this(word.value());
  }
` 
Also, the method setFromString is designed to support setting only the word variable as it can be noticed from the else branch. The bug with the method is that it never goes to the else branch because it passes the first if statement, and crashes at 4th line with StringIndexOutOfBoundsException Exception. This occurs because there are no dividers when we only set a word value. So the first and second variables represent the same value. 

`public void setFromString(String labelStr, String divider) {
    int first = labelStr.indexOf(divider);
    int second = labelStr.lastIndexOf(divider);
    if (first == second) {
      setWord(labelStr.substring(0, first));
      setTag(labelStr.substring(first + 1));
      setLemma(Morphology.lemmaStatic(labelStr.substring(0, first), labelStr.substring(first + 1)));
    } else if (first >= 0) {
      setWord(labelStr.substring(0, first));
      setLemma(labelStr.substring(first + 1, second));
      setTag(labelStr.substring(second + 1));
    } else {
      setWord(labelStr);
      setLemma(null);
      setTag(null);
    }
  }`

Reply:
Ideally this should be fixed then, right?

Reply:
Also, it should be pointed out that commenting out the code isn't necessary... commenting out @Test already disabled the test.

Reply:
Alright, I fixed the underlying bug and restored the test

Reply:
There's a similar bug in the Factory.  Presumably the FactoryTest should be changed to expect null, null, right?  I'll check it in that way, and please confirm that's what you would expect.

Reply:
Yes, there is a similar bug also in Factory. Thank you for solving it.